rakelib/artifacts.rake

```diff
@@ -207,12 +208,26 @@ namespace "artifact" do
     build_docker(true)
   end
 
-  desc "Generate Dockerfile for default and oss images"
+  desc "Generate Dockerfiles for full and oss images"
   task "dockerfiles" => ["prepare", "generate_build_metadata"] do
     puts("[dockerfiles] Building Dockerfiles")
-    build_dockerfiles
+    build_dockerfile(true)
+    build_dockerfile(false)
```

Comment:
Looking at the code here it is a bit confusing to read because it implies that when sending true, you want to build the docker file, but when you send false, you don't want to build it. It doesn't immediately allow the reader to understand that this is to decide whether to build oss/full (Sure, messages kind of allow you to understand but it doesn't make it clear, and someone has to look at the function to really understand what's going on).

I would recommend we change this to defining a type instead of a boolean (e.g. image = oss, image = full) or use numbers to define the type (and obviously document these). Similar to what SELECTED_TEST_SUITE does.

Reply:
This was consistent with how the similar `build_docker` method was implemented, but I agree that in terms of readability it leaves a lot to be desired. I'll change them both

Reply:
I've left the `build_docker` method as is for now. There are changes to come when the `aarch64` flavors are introduced

Reply:
Yes, I think it is best to take care of that in a different PR.